source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -514,9 +514,18 @@ ValidationResults ContextImpl::customVerifyCertChain(
   }
   ASSERT(cert_validator_);
   const char* host_name = SSL_get_servername(ssl, TLSEXT_NAMETYPE_host_name);
+
+  CertValidator::ExtraValidationContext validation_ctx;
+  auto callbacks =
+      static_cast<Network::TransportSocketCallbacks*>(SSL_get_ex_data(ssl, sslSocketIndex()));
+  if (callbacks != nullptr) {
+    validation_ctx.local_address = *callbacks->connection().connectionInfoProvider().localAddress();
```

Comment:
can `localAddress()` return null here?

Reply:
I guess it will not be null but I have changed to shared ptr.

Reply:
My point is that it is always safe to check nullptr before dereferencing a pointer.

Reply:
Yes, there is no dereference since the shared ptr is passed directly